File: modified_scripts/python/matrix_multiplication/gemma_2b.py

def matrix_multiplication(matrix1, matrix2):
    result_matrix = []
    for i in range(len(matrix1)):
        row1 = matrix1[i]
        for j in range(len(matrix2)):
            product = 0
            for k in range(len(matrix2[0])):
                product += row1[k] * matrix2[i][k]
            result_matrix.append(product)
        result_matrix.append([row1[-1] for row1 in matrix1])
    return result_matrix

File: modified_scripts/rust/matrix_multiplication/gemma_2b.rs

fn matrix_multiplication(matrix1: [f32; 3], matrix2: [f32; 3]) -> [f32; 3] {
    
    let mut result_matrix = [0.0; 3];

    
    for i in 0..3 {
      
        for j in 0..3 {
      
            let product = matrix1[i] * matrix2[j];

      
            result_matrix[i][j] += product;
        }
    }

   
    result_matrix
}
